### PR TITLE
Fix end-to-end tests

### PR DESCRIPTION
**Pull request by tristanbrown**
@dblenkus @jkokosar @jvrakor

### CODE REVIEW

File: resdk/tests/functional/analysis/e2e_alignment.py [hunk 1]
```diff
@@ -13,7 +13,7 @@ def test_bowtie2(self):
         reads_1, reads_2 = self.get_reads(2, collection)
         # pylint: enable=unbalanced-tuple-unpacking
         reads_3 = self.get_reads(1)[0]
-        genome = self.get_genome(1)
+        genome = self.get_genome(collection)
```
Comment (jvrakor) [inline resdk/tests/functional/analysis/e2e_alignment.py]: You can also leave this empty as genome does not have to be in a collection.

File: resdk/tests/functional/analysis/e2e_alignment.py [hunk 2]
```diff
@@ -49,7 +49,7 @@ def test_hisat2(self):
...
```
Comment (jvrakor) [inline resdk/tests/functional/analysis/e2e_alignment.py]: Same as above.

File: resdk/tests/functional/analysis/e2e_chip_seq.py [hunk 3]
```diff
@@ -155,6 +135,7 @@ def test_rose(self):
 
         # Normal run on single sample with no background
         rose = macs_1.sample.run_rose2(use_background=False)
+        print(rose[0].input)
```
Comment (jvrakor) [inline resdk/tests/functional/analysis/e2e_chip_seq.py]: This is not needed.

File: resdk/tests/functional/analysis/e2e_expressions.py [hunk 4]
```diff
@@ -14,37 +14,37 @@ def test_cuffquant(self):
         bam_1, bam_2 = self.get_bams(2, collection)
         # pylint: enable=unbalanced-tuple-unpacking
         bam_3 = self.get_bams(1)[0]
-        gff = self.get_gtf(1)
+        gff = self.get_gtf(collection)
```
Comment (jvrakor) [inline resdk/tests/functional/analysis/e2e_expressions.py]: You can also leave this empty as annotation does not have to be in a collection.

File: resdk/tests/functional/analysis/e2e_expressions.py [hunk 5]
```diff
@@ -14,37 +14,37 @@ def test_cuffquant(self):
...
+        self.assertEqual(cuffquants[0].input['annotation'], gff.id)
 
     def test_cuffnorm(self):
         collection = self.res.collection.create(name='Test collection')
 
         # pylint: disable=unbalanced-tuple-unpacking
         cuffquant_1, cuffquant_2, cuffquant_3, cuffquant_4 = self.get_cuffquants(4, collection)
         # pylint: enable=unbalanced-tuple-unpacking
-        gff = self.get_gtf(1)
+        gff = self.get_gtf(collection)
```
Comment (jvrakor) [inline resdk/tests/functional/analysis/e2e_expressions.py]: Same as above.

File: resdk/tests/functional/base.py [hunk 6]
```diff
@@ -93,28 +102,34 @@ def get_reads(self, count=1, collection=None):
             if collection:
                 collection.add_samples(read.sample)
 
+        print("READS GOTTEN")
```
Comment (jvrakor) [inline resdk/tests/functional/base.py]: This is not needed.

File: resdk/tests/functional/base.py [hunk 7]
```diff
@@ -93,28 +102,34 @@ def get_reads(self, count=1, collection=None):
...
+                input={
+                    'src': bam_path,
+                    'species': species,
+                    'build': '02-07-2018'
+                },
                 **collections
             )
             bam.sample.update_descriptor(  # pylint: disable=no-member
-                {'sample': {'organism': 'Homo sapiens'}}
+                {'sample': {'organism': species}}
```
Comment (jvrakor) [inline resdk/tests/functional/base.py]: Why is the descriptor still needed?

**Comment by jvrakor** [comment]
I propose we change the `get_*` functions which have the `build` and `species` inputs to: `def get_genome(self, collection=None, species='Homo sapiens', build='02-07-2018'):` so we have the option to change the inputs. @dblenkus ?

File: resdk/analysis/expressions.py [hunk 8]
```diff
@@ -36,7 +36,7 @@ def cuffquant(resource, gff, genome=None, mask_file=None, library_type=None,
     for sample in get_samples(resource):
         inputs = {
             'alignment': sample.get_bam().id,
-            'gff': get_data_id(gff),
+            'annotation': get_data_id(gff),
```
Comment (jvrakor) [inline resdk/analysis/expressions.py]: You can also change the cuffquant's 'gff' input to 'annotation'.

**Comment by jkokosar** [comment]
@tristanbrown Please fix linter errors. 
@dblenkus Please review

File: resdk/tests/functional/base.py [hunk 9]
```diff
@@ -36,10 +36,11 @@ def setUp(self):
         self.res = Resolwe(ADMIN_USERNAME, ADMIN_PASSWORD, URL)
         self.user_res = Resolwe(USER_USERNAME, USER_PASSWORD, URL)
 
-    def get_genome(self, collection=None):
+    def get_genome(self, collection=None, species='Homo sapiens',
+                    build='02-07-2018'):
```
Comment (dblenkus) [inline resdk/tests/functional/base.py]: Please use a standard genome build, e.g. `hg38`.

File: docs/CHANGELOG.rst [hunk 10]
```diff
@@ -19,6 +19,10 @@ Fixed
   (referended in ``src`` attribute) with no output field
 * Make ``Data.annotation`` an instance attribute instead of class
   attribute
+* Remove annotation tests from ``test_macs``
+* Fix ``get_*`` calls in tests by including species and build inputs
+* Remove/replace invalid collection references in ``get_*`` calls
+* Update cuffquant ``gff`` input to ``annotation`` in helper and test functions
```
Comment (dblenkus) [inline docs/CHANGELOG.rst]: This does not belong into "fix e2e tests" commit. Please move it to the separate commit and mark this change as backward incompatible.

File: resdk/tests/functional/base.py [hunk 11]
```diff
@@ -48,14 +49,19 @@ def get_genome(self, collection=None):
 
         return self.res.run(
             'upload-genome',
-            input={'src': genome_path},
+            input={
+                'src': genome_path,
+                'species': species,
+                'build': build
```
Comment (dblenkus) [inline resdk/tests/functional/base.py]: Add a trailing comma.

File: resdk/tests/functional/base.py [hunk 12]
```diff
@@ -48,14 +49,19 @@ def get_genome(self, collection=None):
...
+            },
             **collections
         )
 
-    def get_gtf(self, collection=None):
+    def get_gtf(self, collection=None, species='Homo sapiens',
+                    build='02-07-2018'):
```
Comment (dblenkus) [inline resdk/tests/functional/base.py]: Change build as described above.

File: resdk/tests/functional/base.py [hunk 13]
```diff
@@ -64,15 +70,20 @@ def get_gtf(self, collection=None):
 
         return self.res.run(
             'upload-gtf',
-            input={'src': gtf_path, 'source': 'NCBI'},
+            input={
+                'src': gtf_path,
+                'source': 'NCBI',
+                'species': species,
+                'build': build
```
Comment (dblenkus) [inline resdk/tests/functional/base.py]: Add a trailing comma.

File: resdk/tests/functional/base.py [hunk 14]
```diff
@@ -152,7 +167,12 @@ def get_cuffquants(self, count=1, collection=None):
         for _ in range(count):
             cuffquant = self.res.run(
                 'upload-cxb',
-                input={'src': cuffquant_path, 'source': 'NCBI'},
+                input={
+                    'src': cuffquant_path,
+                    'source': 'NCBI',
+                    'species': species,
+                    'build': build
```
Comment (dblenkus) [inline resdk/tests/functional/base.py]: Trailing comma.

File: docs/CHANGELOG.rst [hunk 15]
```diff
@@ -19,6 +19,10 @@ Fixed
   (referended in ``src`` attribute) with no output field
 * Make ``Data.annotation`` an instance attribute instead of class
   attribute
+* Remove annotation tests from ``test_macs``
+* Fix ``get_*`` calls in tests by including species and build inputs
+* Remove/replace invalid collection references in ``get_*`` calls
```
Comment (dblenkus) [inline docs/CHANGELOG.rst]: Was this changed only in docstrings? If yes, you can omit the change log.

**Comment by tristanbrown** [comment]
@dblenkus Please review the latest version.

**Comment by jkokosar** [comment]
@mstajdohar I don't have write permissions here.
